### DISCUSSION TITLE
What features are needed inside?

### DISCUSSION THREAD

**Discussion by strijar** [AetherRadio/X6100Control#6] [Ideas]
Now inside the library there are I2C control functions, reading data from a serial port and partially control via gpio. What do we want exactly to be in this library? Do we need to work with sound through PulseAudio? Or something else?

**Comment by ruilvo** [comment]
That's something that we need to decide ourselves.

I don't think we need sound libraries. http://miniaud.io/ is perfect for sound. liquid-dsp is great for, well, DSP. (and I'm thinking about doing AetherDSP eventually. 

I think we should provide API (and examples/docs) for what a person doing a frontend app would need to *fully* control the **backend**, like:

- [ ] Turn on/off wifi/bt
- [ ] Read/write frequency 
- [ ] Select audio input/output of the STM
- [ ] Turn on/off ATU, tune ATU
- [ ] Control the controllable HPF and LPF filters (the "trapezoid" menu)
- [ ] **Read state** of all the above and everything applicable.
- [ ] ?????

You're more aware of what the STM32 is supposed to do than I am, but my idea is to provide controls to control everything it does, without the programmer having to write SPI  or GPIO commands, just calling functions...

**Comment by strijar** [comment]
> That's something that we need to decide ourselves.

I'm talking about how high level a library should be. Now it is very low level. Working with sound does not directly depend on STM32, we take the sound from PulseAudio. But for example, the sound volume or microphone sensitivity setting is controlled via STM32. It turns out the sound should be in one place.

Again, if we plan to work with a Bluetooth headset, then we would like to have it under control in one place too. Connecting, disconnecting, playing audio and intercepting it from the headset.

**Comment by ruilvo** [comment]
More organic discussion in Discord. Interested parties can accompany us there. 

https://discord.gg/tBf6EC2pzw
